src/payment/strategies/adyenv3/adyenv3-payment-strategy.ts

```diff
@@ -0,0 +1,361 @@
+            }
+
+            additionalActionComponent.mount(`#${containerId}`);
+
+            if (onLoad) {
+                onLoad(() => {
+                    reject(new PaymentMethodCancelledError());
+                    additionalActionComponent.unmount();
+                });
+            }
+        });
+    }
+
+    private _mapAdyenPlaceholderData(billingAddress?: BillingAddress): AdyenPlaceholderData {
+        if (!billingAddress) {
+            return {};
+        }
+
+        const {
+            firstName,
+            lastName,
+            address1: street,
+            address2: houseNumberOrName,
+            postalCode,
+            city,
+            stateOrProvinceCode: stateOrProvince,
+            countryCode: country,
+        } = billingAddress;
+
+        return {
+            holderName: `${firstName} ${lastName}`,
+            billingAddress: {
+                street,
+                houseNumberOrName,
+                postalCode,
+                city,
+                stateOrProvince,
+                country,
+            },
+        };
+    }
+
+    private _mountCardVerificationComponent(): Promise<AdyenComponent> {
+        const adyenv3 = this._getPaymentInitializeOptions();
+        const adyenClient = this._getAdyenClient();
+        let cardVerificationComponent: AdyenComponent;
+
+        return new Promise((resolve, reject) => {
+            if (adyenv3.cardVerificationContainerId) {
+                cardVerificationComponent = adyenClient.create(AdyenComponentType.SecuredFields, {
+                    ...adyenv3.options,
+                    styles: {
+                        error: {
+                            color: 'red',
+                        },
+                        validated: {
+                            color: 'green',
+                        },
+                    },
+                    onError: componentState => {
+                        adyenv3.validateCardFields(componentState);
+                    },
+                    onFieldValid: componentState => adyenv3.validateCardFields(componentState),
+                });
+
+                try {
+                    cardVerificationComponent.mount(`#${adyenv3.cardVerificationContainerId}`);
+                } catch (error) {
+                    reject(new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized));
+                }
+            }
+
+            resolve(cardVerificationComponent);
+        });
+    }
+
+    private _mountPaymentComponent(paymentMethod: PaymentMethod): Promise<AdyenComponent> {
+        let paymentComponent: AdyenComponent;
+        const adyenv3 = this._getPaymentInitializeOptions();
+        const adyenClient = this._getAdyenClient();
+
+        return new Promise((resolve, reject) => {
+            const billingAddress = this._store.getState().billingAddress.getBillingAddress();
+
+            paymentComponent = adyenClient.create(paymentMethod.method, {
```

Comment:
Do we need to check if billing address is present?

Reply:
added